apps/oc_chef_authz/src/oc_chef_policy.erl

```diff
@@ -0,0 +1,230 @@
+
+create_record(OrgId, Name, RequestingActorId) ->
+    Policy = #oc_chef_policy{
+                           org_id = OrgId,
+                           name = Name},
+    set_created(Policy, RequestingActorId).
+
+set_created(#oc_chef_policy{} = Object, ActorId) ->
+    Object#oc_chef_policy{last_updated_by = ActorId}.
+
+set_updated(#oc_chef_policy{} = Object, ActorId) ->
+    Object#oc_chef_policy{last_updated_by = ActorId}.
+
+is_indexed() ->
+    false.
+
+ejson_for_indexing(#oc_chef_policy{}, _EjsonTerm) ->
+   {[]}.
+
+update_from_ejson(#oc_chef_policy{} = Policy, PolicyData) ->
+    Name = ej:get({<<"name">>}, PolicyData, name(Policy)),
+    Policy#oc_chef_policy{
+             name = Name,
+             serialized_object = PolicyData
+						 }.
+
+fields_for_update(#oc_chef_policy{
+                     id = Id,
+                     last_updated_by = LastUpdatedBy,
+                     serialized_object = SerializedObject
+                                 } = Policy) ->
+    [LastUpdatedBy, name(Policy), policy_group(Policy), chef_db_compression:compress(oc_chef_policy, jiffy:encode(SerializedObject)), Id].
+
+
+fields_for_fetch(#oc_chef_policy{org_id = OrgId} = Policy) ->
+    [name(Policy), policy_group(Policy), OrgId].
+
+record_fields() ->
+    record_info(fields, oc_chef_policy).
+
+list(#oc_chef_policy{org_id = OrgId}, CallbackFun) ->
+    CallbackFun({list_query(), [OrgId], rows}).
+
+update(#oc_chef_policy{
+                      org_id = _OrgId,
+                      authz_id = _PolicyAuthzId,
+                      last_updated_by = _AuthzId
+                     } = Record, CallbackFun) ->
+	chef_object:default_update(Record, CallbackFun).
+
+
+parse_binary_json(Bin) ->
+    Policy = chef_json:decode_body(Bin),
+    case ej:valid(?VALIDATION_CONSTRAINTS, Policy) of
+        ok ->
+			{ok, Policy};
+        Bad ->
+            throw(Bad)
+    end.
+
+
+fetch(#oc_chef_policy{} = Record, CallbackFun) ->
+    chef_object:default_fetch(Record, CallbackFun).
+
+
+flatten(#oc_chef_policy{
+          id = Id,
+          authz_id = AuthzId,
+          org_id = OrgId,
+          name = Name,
+          policy_group = Group,
+          last_updated_by = LastUpdatedBy,
+          serialized_object = SerializedObject}) ->
+	Compressed = chef_db_compression:compress(oc_chef_policy, jiffy:encode(ej:delete({<<"policy_group">>}, SerializedObject))),
+    [Id, AuthzId, OrgId, Name, Group, LastUpdatedBy, Compressed].
+
+
+assemble_policy_ejson(#oc_chef_policy{
+                         serialized_object = SerializedObject
+                        }, _OrgName) ->
+    jiffy:decode(chef_db_compression:decompress(SerializedObject)).
+
+delete(ObjectRec = #oc_chef_policy{
+                      org_id = OrgId,
+                      last_updated_by = _AuthzId,
+                      authz_id = _PolicyAuthzId
+                     }, CallbackFun) ->
+    CallbackFun({delete_query(), [name(ObjectRec), policy_group(ObjectRec), OrgId]}).
```

Comment:
Just a general question, and not related to this code, but after implementing some more `chef_objects` with this refactor, how does it feel? Are there any future improvements that we could make that might make life easier? For @wk8, being new to this code base, did this make sense?